app_build_suite/build_steps/base_test_runner.py

```diff
@@ -395,6 +402,8 @@ def _upload_chart_to_app_catalog(self, context: Context):
 
     # noinspection PyMethodMayBeStatic
     def _delete_app(self, config: argparse.Namespace, context: Context):
+        if get_config_value_by_cmd_line_option(config, BaseTestRunnersFilteringPipeline.key_config_option_deploy_app):
```

Comment:
Wait, if that option was there, we should delete it, not `return`, right?

Reply:
ah should be `if not ...`?

This is somehow related to giantswarm/giantswarm#15302

Reply:
Yeah, it seems so. We could really use some tests around this logic.

Reply:
```suggestion
        if not get_config_value_by_cmd_line_option(config, BaseTestRunnersFilteringPipeline.key_config_option_deploy_app):
```